apps/oc_chef_authz/src/oc_chef_policy_group.erl

```diff
@@ -0,0 +1,180 @@
+
+id(#oc_chef_policy_group{id = Id}) ->
+    Id.
+
+name(#oc_chef_policy_group{name = Name}) ->
+    Name.
+
+org_id(#oc_chef_policy_group{org_id = OrgId}) ->
+    OrgId.
+
+type_name(#oc_chef_policy_group{}) ->
+    policy.
+
+authz_id(#oc_chef_policy_group{authz_id = AuthzId}) ->
+    AuthzId.
+
+
+create_query() ->
+    insert_policy_group.
+
+update_query() ->
+    update_policy_group_by_id.
+
+delete_query() ->
+    delete_policy_group_by_id.
+
+find_query() ->
+    find_policy_group_by_orgid_name.
+
+list_query() ->
+    list_policy_groups_for_org.
+
+bulk_get_query() ->
+    %% TODO: do we need this?
+    ok.
+
+new_record(OrgId, AuthzId, PolicyGroupData) ->
+    Name = ej:get({<<"name">>}, PolicyGroupData),
+    Id = chef_object_base:make_org_prefix_id(OrgId, Name),
+    #oc_chef_policy_group{
+        id = Id,
+        authz_id = AuthzId,
+        org_id = OrgId,
+        name = Name}.
+
+create_record(OrgId, Name, RequestingActorId) ->
+    Policy = #oc_chef_policy_group{
+                           org_id = OrgId,
+                           name = Name},
+    set_created(Policy, RequestingActorId).
+
+set_created(#oc_chef_policy_group{} = Object, ActorId) ->
+    Object#oc_chef_policy_group{last_updated_by = ActorId}.
+
+set_updated(#oc_chef_policy_group{} = Object, ActorId) ->
+    Object#oc_chef_policy_group{last_updated_by = ActorId}.
+
+is_indexed() ->
+    false.
+
+ejson_for_indexing(#oc_chef_policy_group{}, _EjsonTerm) ->
+   {[]}.
+
+update_from_ejson(#oc_chef_policy_group{} = Policy, PolicyGroupData) ->
+    Name = ej:get({<<"name">>}, PolicyGroupData, name(Policy)),
+    Policy#oc_chef_policy_group{name = Name}.
+
+fields_for_update(#oc_chef_policy_group{
+                     id = Id,
+                     last_updated_by = LastUpdatedBy
+                                 } = PolicyGroup) ->
+    [LastUpdatedBy, name(PolicyGroup), Id].
+
+
+fields_for_fetch(#oc_chef_policy_group{org_id = OrgId} = PolicyGroup) ->
+    [name(PolicyGroup), OrgId].
+
+record_fields() ->
+    record_info(fields, oc_chef_policy_group).
+
+list(#oc_chef_policy_group{org_id = OrgId}, CallbackFun) ->
+    CallbackFun({list_query(), [OrgId], rows}).
+
+update(#oc_chef_policy_group{
+                      org_id = _OrgId,
+                      authz_id = _PolicyGroupAuthzId,
+                      last_updated_by = _AuthzId
+                     } = Record, CallbackFun) ->
+	chef_object:default_update(Record, CallbackFun).
+
+
+parse_binary_json(Bin) ->
+    PolicyGroup = chef_json:decode_body(Bin),
+    case ej:valid(?VALIDATION_CONSTRAINTS, PolicyGroup) of
+        ok ->
+			{ok, PolicyGroup};
+        Bad ->
+            throw(Bad)
+    end.
+
+flatten(#oc_chef_policy_group{
+          id = Id,
+          authz_id = AuthzId,
+          org_id = OrgId,
+          name = Name,
+          last_updated_by = LastUpdatedBy}) ->
+    [Id, AuthzId, OrgId, Name, LastUpdatedBy].
```

Comment:
Looks like it should be possible to remove this and use the default flatten since it grabs all fields of the record in order?